core/error.rng

```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<grammar ns="urn:ietf:rfc:7807" xmlns="http://relaxng.org/ns/structure/1.0"
+    datatypeLibrary="http://www.w3.org/2001/XMLSchema-datatypes">
+
+    <start>
+
+        <element name="problem">
+
+            <zeroOrMore>
+                <ref name="foreign-attribute"/>
+            </zeroOrMore>
+
+            <interleave>
+
+                <element name="status">
+
+                    <zeroOrMore>
+                        <ref name="foreign-attribute"/>
+                    </zeroOrMore>
+
+                    <data type="integer">
+                        <param name="minInclusive">400</param>
```

Comment:
Is the ``error`` model only for returning errors in HTTP connections? And, would all API endpoints return these as the body of the response of an HTTP status code?

Or, would ``error`` also represent internal validation errors?

Reply:
Described in https://docs.google.com/document/d/1JhVpW8dxAuJqnk2YsQa76yxdIV_BlOd1QIi02fIu9mo/edit?usp=sharing (required format for any `400`-`599` response.)

Reply:
I think the intention is for the API endpoints to return this, simplest possible case is a 404. If you apply it to a `PUT /article/:id/version/:v` it could contain a validation error, but it could probably use a custom tag to be expressed.

Internally, you would keep an exception around (with some data structures inside it if the error is very complex) but you would only need to serialize it to XML when returning the response.

Reply:
I see, this is not my area. Quick read of rfc7807 and an example there doesn't require ``status`` in the XML, however it seems reasonable to also place it there for consistency :+1:

Reply:
I don't like that it doesn't actually require anything. `status` and `title` should be, as the response already has a status line so can duplicate that as a minimum for easier processing. Try to make this compliant, but a little stricter.

Reply:
I guess there's no easy way to makes sure the HTTP status code also matches the XML ``status`` value, would have to be enforced higher up from the relax ng schemas.